src/app/services/upload/upload-builder.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { FileBuilderService } from '../file-builder/file-builder.servise';
+import { RequestService } from '../request/request.service';
+import { SigningResponse } from '../signing/signing-response.interface';
+import { UploadResult } from './upload-result.interface';
+
+@Injectable()
+export class UploadBuilderService {
+  constructor(private requestService: RequestService, private fileBuilderService: FileBuilderService) {}
+
+  uploadToAmazon(responseJSON: SigningResponse, uploadResult: UploadResult): Observable<UploadResult> {
+    const formData = new FormData();
+
+    this.buildForm(formData, responseJSON);
+    formData.append('file', uploadResult.file);
+
+    uploadResult.filePath = this.fileBuilderService.getFilePath(uploadResult, responseJSON.file_url_path);
+    uploadResult.publicUrl = this.fileBuilderService.getPublicUrl(uploadResult, responseJSON.file_prefix);
```

Comment:
After a successful response, it combine the response `file_url_path` with the result `filePath`